ecosystem/web-wallet/src/core/components/TransferDrawer.tsx

```diff
@@ -0,0 +1,289 @@
+import useWalletState from 'core/hooks/useWalletState';
+import {
+  accountExists,
+  getAccountResources,
+  getTestCoinTokenBalanceFromAccountResources,
+  getToAddressAccountExists,
+} from 'core/queries/account';
+import { submitTestCoinTransferTransaction } from 'core/mutations/transaction';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { getUserTransaction } from 'core/queries/transaction';
+import { ExternalLinkIcon } from '@chakra-ui/icons';
+import {
+  NODE_URL,
+  secondaryErrorMessageColor, STATIC_GAS_AMOUNT,
+} from 'core/constants';
+import numeral from 'numeral';
+import { GraceHopperBoringAvatar } from 'core/components/BoringAvatar';
+import { secondaryTextColor } from '../../pages/Login';
+
+export const secondaryDividerColor = {
+  dark: 'whiteAlpha.300',
+  light: 'gray.200',
+};
+
+function TransferDrawer() {
+  const { colorMode } = useColorMode();
+  const { aptosAccount } = useWalletState();
+  const toast = useToast();
+  const queryClient = useQueryClient();
+  const {
+    formState: { errors }, handleSubmit, register, setError, watch,
+  } = useForm();
+  const addressInputRef = useRef<HTMLInputElement>();
+  const { isOpen, onClose, onOpen } = useDisclosure();
+  const {
+    isLoading: isTransferLoading,
+    mutateAsync: submitSendTransaction,
+  } = useMutation(submitTestCoinTransferTransaction, {
+    onSettled: async (txnHash) => {
+      if (!txnHash) {
+        return;
+      }
+      queryClient.invalidateQueries('getAccountResources');
+      const txn = await getUserTransaction({ txnHashOrVersion: txnHash });
+      const amount = (txn?.payload)
+        ? (txn.payload as { arguments: string[] }).arguments[1]
+        : undefined;
+      toast({
+        description: (txn?.success) ? `Amount transferred: ${amount}, gas consumed: ${txn?.gas_used}` : `Transfer failed, gas consumed: ${txn?.gas_used}`,
+        duration: 5000,
+        isClosable: true,
+        status: (txn?.success) ? 'success' : 'error',
+        title: `Transaction ${txn?.success ? 'success' : 'error'}`,
+        variant: 'solid',
+      });
+    },
+  });
+
+  const transferAmount: string | undefined | null = watch('transferAmount');
+  const transferAmountNumeral = numeral(transferAmount).format('0,0');
+
+  const {
+    onChange: addressOnChange,
+    ref: toAddressRef,
+    ...toAddressRest
+  } = { ...register('toAddress') };
+  const toAddress: string | undefined | null = watch('toAddress');
+  const explorerAddress = `https://explorer.devnet.aptos.dev/account/${toAddress}`;
+  const { data: toAddressAccountExists } = useQuery(
+    ['getToAddressAccountExists', { aptosAccount, toAddress }],
+    getToAddressAccountExists,
+  );
+
+  const onSubmit: SubmitHandler<Record<string, any>> = async (data, event) => {
+    event?.preventDefault();
+    if (toAddress && aptosAccount && transferAmount) {
+      const toAccountExists = await accountExists({ address: toAddress });
```

Comment:
Is it possible to reuse the `toAddressAccountExists`?

If not, can these requests be made in parallel instead of one after the other? `const [toAccountExists, fromAccountResources]  = await Promise.all([accountExists(), getAccountResources()])`

Reply:
I use `toAddress` because that's used in line 105 when i provide the address to the `accountExists` function (it doesn't take undefined `toAddress`. 

We could do them in parallel but they are different errors and I'd rather have it stop immediately before proceeding if there's an error. As of right now, I think the response latency is pretty fast, but I do agree that stringing this many awaits together is not ideal. I will try to make this better if time allows.